### PR TITLE
upgrade the dependent version of ecto

### PR DESCRIPTION
**Pull request by sbatson5**
Upgrade our dependent version of ecto.

### CODE REVIEW

File: test/valid_field_test.exs [hunk 1]
```diff
@@ -24,7 +24,7 @@ defmodule ValidFieldTest do
       |> ValidField.assert_invalid_field(:title, ["", nil, "Something else"])
     end
 
-    assert_raise ValidField.ValidationError, "Expected the following values to be invalid for \"date_of_birth\": #Ecto.Date<2016-08-23>", fn ->
+    assert_raise ValidField.ValidationError, "Expected the following values to be invalid for \"date_of_birth\": #Ecto.Date<#{get_current_date}>", fn ->
```
Comment (sbatson5) [inline test/valid_field_test.exs]: @BlakeWilliams 👋 
I was looking at https://github.com/DockYard/valid_field/pull/21/files and wasn't sure why I had to make this change to get the tests to pass.

Comment (bcardarella) [inline test/valid_field_test.exs]: time based testing is a PITA

File: mix.exs [hunk 2]
```diff
@@ -44,7 +44,7 @@ defmodule ValidField.Mixfile do
 
   defp deps do
     [
-      {:ecto, "~> 2.0", only: [:dev, :test]},
+      {:ecto, "~> 2.1.2", only: [:dev, :test]},
```
Comment (bcardarella) [inline mix.exs]: can you change this to `~> 2.1`

File: test/support/model.ex [hunk 3]
```diff
@@ -8,17 +8,15 @@ defmodule ValidField.Support.Model do
...
     timestamps
   end
 
-  @required_fields ~w(first_name)
-  @optional_fields ~w(last_name title password password_confirmation)
-
   def changeset(model, params \\ :empty) do
     model
-    |> cast(params, @required_fields, @optional_fields)
+    |> cast(params, [:first_name, :last_name, :title, :password, :password_confirmation, :date_of_birth])
```
Comment (bcardarella) [inline test/support/model.ex:18]: why this change?

File: test/valid_field_test.exs (hunk 1 above)
Comment (bcardarella) [reply test/valid_field_test.exs]: if `get_current_date/0` is not being used elsewhere you should just interpolate on `Ecto.Date.utc()`

Comment (bcardarella) [reply test/valid_field_test.exs]: FWIW, the longer-term fix is to disassociate the datetime functionality into something that can be stubbed

**Review by bcardarella — CHANGES_REQUESTED** [review]

File: test/support/model.ex (hunk 3 above)
Comment (sbatson5) [reply test/support/model.ex:18]: `cast/4` is being deprecated in favor of `cast/3` with `validate_required`.  Our required fields should be listed in `validate_required`

Comment (sbatson5) [reply test/support/model.ex:18]: We could still list the fields above: `@fields w(first_name, last_name, title ...)a` but this is the convention I've seen more often.

Comment (brzpegasus) [reply test/support/model.ex:18]: That explains what I saw on Mike North's changeset prezo last week.

Comment (sbatson5) [reply test/support/model.ex:18]: Are we happy to mark this as resolved?

**Review by bcardarella — APPROVED** [review]

**Pull request merged**